api-operator/pkg/controller/api/api_controller.go

```diff
@@ -515,13 +515,29 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 			return reconcile.Result{}, errMgwSvc
 		}
 
-		// create horizontal pod auto-scalar
-		hpa := mgw.HPA(instance, mgwDeployment, ownerRef)
-		if errHpa := k8s.CreateIfNotExists(&r.client, hpa); errHpa != nil {
-			reqLogger.Error(errHpa, "Error creating the horizontal pod auto-scalar", "hpa_name", hpa.Name)
+		// get global hpa configs, return error if not found (required config map)
+		hpaConfMap := k8s.NewConfMap()
+		errHpa := k8s.Get(&r.client, types.NamespacedName{Namespace: wso2NameSpaceConst, Name: hpaConfigMapName}, hpaConfMap)
+		if errHpa != nil {
+			reqLogger.Error(errHpa, "HPA configs not defined")
 			return reconcile.Result{}, errHpa
 		}
 
+		// create horizontal pod auto-scalar
+		hpaV2beta1, hpaV2beta2 := mgw.HPA(&r.client, instance, mgwDeployment, ownerRef)
+		if hpaConfMap.Data[hpaVersionConst] == "v2beta1" {
+			if errHpaV2beta1 := k8s.CreateIfNotExists(&r.client, hpaV2beta1); errHpaV2beta1 != nil {
+				reqLogger.Error(errHpaV2beta1, "Error creating the horizontal pod auto-scalar with HPA version v2beta1", "hpa_name", hpaV2beta1.Name)
+				return reconcile.Result{}, errHpaV2beta1
+			}
+		}
+		if hpaConfMap.Data[hpaVersionConst] == "v2beta2" {
+			if errHpaV2beta2 := k8s.CreateIfNotExists(&r.client, hpaV2beta2); errHpaV2beta2 != nil {
+				reqLogger.Error(errHpaV2beta2, "Error creating the horizontal pod auto-scalar with HPA version v2beta2", "hpa_name", hpaV2beta2.Name)
+				return reconcile.Result{}, errHpaV2beta2
+			}
+		}
```

Comment:
We dont need to check the versions. For given version in the config, we can set this here.

Reply:
Does v2beta1 works in K8s 1.14?

Reply:
Fixed by 831fcc26a11577c5cd7a937ea772b5d4b48c6921.
v2beta1 works in k8s 1.14.